pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -32,6 +32,16 @@ type httpScrapeClient struct {
 	httpClient *http.Client
 }
 
+// Allocate a large enough buffer to handle the maximum
+// marshalled Stat size which is 292 bytes.
+const PoolSize = 320
```

Comment:
Not sure if we can calculate this in a more dynamic way @vagababov.

Reply:
We could put a dummy `Stat` next to it and call the `Size()` function on it maybe?

Reply:
tbh think we might be best just putting a bytes.Buffer in the pool.. looking at https://github.com/golang/go/issues/23199 the problem really only occurs if the things you put in the byte buffer that have significantly variable sizes, but here they're all guaranteed to be < 300ish bytes so :shrug:.

Reply:
(that also avoids the allocation we'd otherwise have to make to put a slice into an interface{})

Reply:
@julz when I tried the buffer thing I was seeing capacity increasing in re-used buffers which made me believe that we are hitting the same problem, independently of the size of the memory used. 
Right now we are on the safe side. However, I could benchmark that option too.

Reply:
yeah.. Id love to understand why that happened, if so. Maybe worth trying it quickly just to check, if it works it'd be nicer

Reply:
@markusthoemmes size does not add the string size  which can be unbounded btw :(
It only adds 16 bytes for the string ref afaik.

Reply:
the string is a pod name right? I think that has max 253 chars if so

Reply:
@julz I was refering to the Size func, but yes it is the K8s maximum.

Reply:
right, but it looks like https://github.com/knative/serving/blob/master/pkg/autoscaler/metrics/stat.pb.go#L442 would include the podName size, which you could just statically set in the dummy stat to a 253-length string. (I still think bytes.Buffer is better :D).

Reply:
Well it kinda needs to have the "proper" size as it's used by the `Marshal` machinery to size the buffer accordingly :yum:

Reply:
I agree that `bytes.Buffer` makes things easier though as we don't have to worry about it as much.